cmd/mesh/manifest.go

```diff
@@ -89,6 +93,27 @@ func genManifests(inFilename string, setOverlayYAML string) (name.ManifestMap, e
 	if err != nil {
 		return nil, err
 	}
+	// for installation path contains version number only
+	isp := mergedICPS.InstallPackagePath
+	if version.IsVersionString(isp) {
+		mergedICPS.InstallPackagePath = fmt.Sprintf(helm.InstallationPathTemplate, isp, isp)
+	}
+	if util.IsHTTPURL(mergedICPS.InstallPackagePath) {
+		uf, err := helm.NewURLFetcher(mergedICPS.InstallPackagePath, "")
```

Comment:
define a temp dir here since we want subsequent calls to this command to use the same dir.

Reply:
it is actually using the same directory instead of creating new ones with random prefix. You can check the logic at NewURLFetcher. To avoid confusion, I will update the naming and comment.